driver-core/src/main/com/mongodb/client/model/expressions/NumberExpression.java

```diff
@@ -21,4 +21,37 @@
  */
 public interface NumberExpression extends Expression {
 
+    NumberExpression multiply(NumberExpression n);
+
+    default NumberExpression multiply(final Number multiply) {
+        return this.multiply(Expressions.numberToExpression(multiply));
+    }
+
+    NumberExpression divide(NumberExpression n);
+
+    default NumberExpression divide(final Number divide) {
+        return this.divide(Expressions.numberToExpression(divide));
+    }
+
+    NumberExpression add(NumberExpression n);
+
+    default NumberExpression add(final Number add) {
+        return this.add(Expressions.numberToExpression(add));
+    }
+
+    NumberExpression subtract(NumberExpression n);
+
+    default NumberExpression subtract(final Number subtract) {
+        return this.subtract(Expressions.numberToExpression(subtract));
+    }
+
+    NumberExpression max(NumberExpression n);
```

Comment:
Why don't `min` and `max` have `Number` overloads?

Reply:
Uncertain. It seems that, if an overload is useful at all, it's useful on multiply. But then there's `or(BoolEx)` where an overload seems inappropriate. I'm not sure where to draw the line formally - to me it seemed odd to `max` on a known number.

When I thought about an overload on `max`, it seemed that `max` should be renamed to `lowerBound` (that is, if your n is under 5, `n.lowerBound(5)` would return 5, and n otherwise).

Reply:
ok, we can always add more overloads in the future.